### ISSUE TITLE
General questions regarding hyper parameter.

### ISSUE THREAD

**Issue by AaronSpieler** [White-Link/UnsupervisedScalableRepresentationLearningTimeSeries#10]
Hello,

Thank you for your great paper!

During application of your software some questions arose to me:

1) Have you tried tuning the `gamma` parameter of the SVM? (paper only mentions `C`)
2)The paper recommends 1500-2000 optimization steps, however, what if my dataset has 40k time series
   2.1) Any recommendation on how to set/tune the number of optimization steps?
3) In the paper representations of length 80-160 were trained, however, to me that seems a quite long, given that many datasets in UCR have time series shorter than that. 
   3.1) Have you done any test with 16-32 dimensional representations? 
   3.2) Have you examined the correlation between performance and underlying dataset avg. time series length?
   3.3) Any recommendations on lower bound on dimensionality?

Any insight is highly appreciated!

**Comment by White-Link** [comment]
Hi, thank you for your interest in our work!

You questions mainly deal with the hyperparameters that we chose for our unsupervised method. Since we were working with this fully unsupervised encoding method on a large number of datasets, most of them being quite small, we wanted to make sure that these hyperparameters could fit most of them, explaining these choices.

More specifically:

1. We did not extensively tune most of the SVM parameters. We kept `gamma` to its default value on scikit-learn v0.20.0.
2. Note that, when K (the number of negative samples in the loss) is equal to 10, 2000 optimization steps corresponds to approximately 20k samples seen by the network, so it could cover half of your dataset. Your dataset being significantly larger than most, if not all, datasets that we study in our article, you might indeed want to increase the number of optimization steps. A good way to choose it, if you have a validation task (e.g., regression or classification), would be to evaluate its performance with respect to the number of optimization steps.
3. Indeed, some UCR datasets have time series which are shorter than our representations. However, these static representations should capture diverse temporal information such as short and long-term dependencies, which might require a large number of dimensions. We chose a representation size that we believe to be large enough for most studied datasets, but you may probably reduce it without significant performance decrease. We did not run a lot of tests regarding the size of the representations, so it is unfortunately difficult to answer your questions precisely. We are of course welcoming any feedback!

I hope this answers your questions. Let us know if you have any other one!

**Comment by AaronSpieler** [comment]
Your response answers all my questions, and is consistent with what I understood from you paper.

Thank you for taking the time, it's highly appreciated!

**Comment by AaronSpieler** [comment]
I might post some insights here regarding representation dimensionality given avg. time series length & number of optimisation steps given dataset size should I get some.
